php/libraries/NDB_Caller.class.inc

```diff
@@ -94,7 +94,7 @@ class NDB_Caller
         ) {
             header("HTTP/1.1 303 See Other");
             $url  = $settings->getBaseURL();
-            $url .= Utility::removeCommonAffix($url, $_SERVER['REQUEST_URI']);
+            $url .= parse_url($_SERVER['REQUEST_URI'], PHP_URL_PATH);
```

Comment:
These don't seem to do the same thing at all.

One removes any common prefix/suffix to make a path relative to the LORIS root's baseurl  and make it relative, the other just parses the absolute URL given by PHP.

Reply:
But is this code even used at all? the login endpoint goes through the index.php now and NDB_Caller only exists for instruments (which don't have a module to route to yet) and surveys (which don't have a login)

Reply:
I'm not sure. You'd know better than I would. Should I try deleting this section of the code?

Reply:
I think the only impact would be for a session timeout on an instrument page.

Reply:
What would happen in that case? A user would not be properly redirected to the login page? Let me know what approach you'd like to take here.

Reply:
I'm not very familiar with this code, but I think what it's *trying* to do is redirect to the page that you were on when it notices a login attempt from a session timeout on a non-login page.

The approach I'd like to take is to eventually kill NDB_Caller entirely since our router does the same thing in a more robust way. Maybe we can just move removeCommonAffix into NDB_Caller since it's the only place that uses it and not put much effort into changing things that have the potential to break obscure code paths there